Cookbook/Technical-Documents/XcodeTips.md

```diff
@@ -1,20 +1,70 @@
-## Tips and tricks
+# Xcode tips and tricks
 
-### Errors when pulling code/switching branches
+## Keybindings
+As we spend so much time in our ~~favourite~~ editor, it's essential to learn keybindings to decrease content switching and maintain focus.
+
+#### Navigating and finding things
+As long as the project is indexed, Xcode lets you jump to anything from anywhere.
+
+- **Cmd-Shift-0** to open the Apple documentation window.
+- **Cmd-Ctrl-[** or **]** to cycle between targets.
+
+##### Find symbols, files or text
+- **Cmd-Shift-O** and type anything to quickly jump to files or symbols. **Enter** to open, **Opt-Enter** to open in the assistant editor.
+- **Cmd-Ctrl-J** (as in “jump”) or **Cmd-Ctrl-Click** on a symbol to go to definition. Add **Opt** to open in assistant editor.
+- Right click and “Find Selected Symbol in Workspace” to show in the search pane where the symbol under cursor is used.
+- **Cmd-Shift-F** to search in workspace (the whole project). You'll want to select Text on top because searching for references is tricky manually, they require specific syntax.
+- **Cmd-Opt-J** and type to filter files by name in the Project Navigator (or tests in the Test Navigator). Useful if you want to find multiple files with the same name, or can't quite remember how a file is called.
+
+##### File level navigation
+- **Cmd-Shift-J** to show the open file in the Project Navigator.
+- **Cmd-Ctrl-Left** or **Right** in a file to navigate between recently open files.
+- **Ctrl-6** in a file to open the list of methods and variables. You can then type to apply a filter and hit Enter to jump to the selected completion.
+- **Cmd-Opt-Shift-Left** in a file to fold all methods and **Cmd-Opt-Shift-Right** to unfold. May be useful to get a quick overview.
+
+#### Making space for your code
+Use these keybindings to give your code more space, and show the panes again quickly when needed. You can keep the panes nice and wide this way, and just hide them when you're not using them, rather than keep them thin and see them all the time.
+
+- **Cmd-Shift-Y** to show or hide the bottom drawer (locals and console output).
+- **Cmd-0** and **Cmd-Opt-0** to fold and unfold the left and right panes, respectively.
+- **Cmd-Enter** to close the assistant editor and leave only the main editor pane.
```

Comment:
I use custom behaviors a lot to switch for entire setups, I highly recommend it.

Basically in Xcode menu > Behaviors > Edit Behaviors you can add custom behaviors telling which pane and editors to show and hide, then assign a custom shortcut on that, and now you can switch for example to "max space to edit my code with only main editor without the project navigator nor inspector nor bottom drawer" just with one shortcut

Reply:
Example to have Ctrl-Option-Cmd-D toggle my panes properly ready for a debug session the way I like it

![image](https://user-images.githubusercontent.com/216089/62640920-dab84400-b942-11e9-9e53-f32e09fb7159.png)

Reply:
Very nice Olivier, for some reason I never explored this tab 🤦‍♂ I'll be sure to mention it in the next pass of addressing on this.